pyxem/tests/test_workflows/test_orientation_mapping_phys.py

```diff
@@ -96,13 +96,28 @@ def generate_diffraction_patterns(structure, edc):
     dp = pxm.ElectronDiffraction2D([[pattern, pattern], [pattern, pattern]])
     return dp
 
+def generate_difficult_diffraction_patterns(structure, edc):
+    # generates 4, dumb patterns in a 2x2 array, rotated around c axis
+    difficult_diffraction_half_side_length = 73
+    dp_library = get_template_library(structure, [(90, 90, 4)], edc)
+    for sim in dp_library["A"]["simulations"]:
+        pattern = sim_as_signal(sim, 2 * difficult_diffraction_half_side_length, 0.025, 1).data
+    dp = pxm.ElectronDiffraction2D([[pattern, pattern], [pattern, pattern]])
+    return dp
+
 
 def get_template_match_results(structure, edc, rot_list, mask=None):
     dp = generate_diffraction_patterns(structure, edc)
     library = get_template_library(structure, rot_list, edc)
     indexer = IndexationGenerator(dp, library)
     return indexer.correlate(mask=mask, method="zero_mean_normalized_correlation")
 
+def get_template_match_results_fullframe(structure, edc, rot_list, mask=None):
+    dp = generate_diffraction_patterns(structure, edc)
+    library = get_template_library(structure, rot_list, edc)
+    indexer = IndexationGenerator(dp, library)
+    return indexer.correlate(mask=mask, method="full_frame_correlation")
+
```

Comment:
I would add a new test, `get_template_match_results_fullframe_bad_size()` which you use a `pytest.mark.xfail()` on which runs the code as above, except with the `dp` being made with `generate_difficult_diffraction_pattern()`